### PR TITLE
Fixed issue for scenario where handleClickOutside does not get called

### PR DESCRIPTION
**Pull request by jcursoli**
Fixed issue for when manually passing in disableOnClickOutside and preventing handleClickOutside from getting called.

### CODE REVIEW

**Comment by jcursoli** [comment]
@Pomax Is there anything I can do to make this PR move a bit faster? Just asking because we use this heavily in our app. Thanks in advance!

**Comment by Pomax** [comment]
If you can add a test that fails without your fix, and passes with your fix, that'll speed things up a lot because it means review goes from "figuring out what is going on", which is a lengthy process, to simply "confirming your test case", which takes almost no time at all.

**Comment by jcursoli** [comment]
@Pomax Okay its been a while but i added the test cases

**Comment by Pomax** [comment]
perfect, I'll have a look at this as soon as I can!

**Comment by Pomax** [comment]
Confirmed that the tests did not work with current, and work with your fix, so: in it goes.

**Comment by jcursoli** [comment]
Thanks a ton!

File: dist/react-onclickoutside.js [hunk 1]
```diff
@@ -293,7 +292,9 @@ function onClickOutsideHOC(WrappedComponent, config) {
         }
       }
 
-      this.componentNode = reactDom.findDOMNode(this.getInstance());
+      this.componentNode = reactDom.findDOMNode(this.getInstance()); // return early so we dont initiate onClickOutside
+
+      if (this.props.disableOnClickOutside) return;
```
Comment (tomasztomys) [inline dist/react-onclickoutside.js]: onclickoutside will never initialize if disableOnClickOutside will update (from true to false or from false to true)

Comment (tomasztomys) [reply dist/react-onclickoutside.js]: @Pomax

Comment (jcursoli) [reply dist/react-onclickoutside.js]: Im trying to understand what you are saying, but isnt that the point of passing disable so it does not get initialised. and if you add it to componentDidUpdate, the handleclickoutside will be initialised.

Comment (tomasztomys) [reply dist/react-onclickoutside.js]: If you pass disabled as true and then you change props disabled to false (in component in your app), outside will not be initialized, so will not work..

Comment (jcursoli) [reply dist/react-onclickoutside.js]: a quick solution would be to return early in your handclickoutside if your prop is disabled. reverting this commit will make it so its never disabled.

**Pull request merged**